desc/objectives/linear_objectives.py

```diff
@@ -39,6 +40,223 @@ def update_target(self, eq):
+    normalize : bool, optional
+        Whether to compute the error in physical units or non-dimensionalize.
+        Has no effect for this objective.
+    normalize_target : bool, optional
+        Whether target and bounds should be normalized before comparing to computed
+        values. If `normalize` is `True` and the target is in physical units,
+        this should also be set to True. Has no effect for this objective.
+    name : str, optional
+        Name of the objective function.
+
+    """
+
+    _scalar = False
+    _linear = True
+    _fixes = True
+    _units = "(~)"
+    _print_value_fmt = "Fixed parameter error: {:10.3e} "
+
+    def __init__(
+        self,
+        thing=None,
+        params=None,
+        target=None,
+        bounds=None,
+        weight=1,
+        normalize=False,
+        normalize_target=False,
+        name="Fixed parameter",
+    ):
+
+        self._target_from_user = target
+        self._params = params
+        super().__init__(
+            things=thing,
+            target=target,
+            bounds=bounds,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            name=name,
+        )
+
+    def build(self, thing=None, use_jit=False, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        self.things = setdefault(thing, self.things)
+        thing = self.things[0]
+        params = setdefault(self._params, thing.optimizable_params)
+
+        if not isinstance(params, (list, tuple)):
+            params = [params]
+        assert all(par in thing.opimizable_params for par in params)
+        self._params = params
+        target = setdefault(
+            self._target_from_user, {par: thing.params[par] for par in params}
+        )
+        self._dim_f = sum(thing.dimensions[p] for p in params)
+        self.target = jnp.concatenate([target[par] for par in params])
+
+        super().build(things=thing, use_jit=use_jit, verbose=verbose)
+
+    def compute(self, params, constants=None):
+        """Compute fixed degree of freedom errors.
+
+        Parameters
+        ----------
+        params : dict
+            Dictionary of equilibrium degrees of freedom, eg Equilibrium.params_dict
+        constants : dict
+            Dictionary of constant data, eg transforms, profiles etc. Defaults to
+            self.constants
+
+        Returns
+        -------
+        f : ndarray
+            Fixed degree of freedom errors.
+
+        """
+        return jnp.concatenate([params[par] for par in self._params])
+
+
+class ShareParameter(_Objective):
+    """Require that params from two or more Optimizable objects share the same values.
```

Comment:
Clarifying question:
So by default, every optimizable "thing" has its own copy of optimizable parameters, right? Like if I am optimizing two coils, they are completely independent. But then, for example, I could use this objective to say they have to have the same position, while letting their other degrees of freedom (tilt, rotation, etc.) be independent?

Reply:
That's the idea yes